arch/arm/include/aarch32/cortex_a_r/exc.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2018 Lexmark International, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Exception/interrupt context helpers for Cortex-A and Cortex-R CPUs
+ *
+ * Exception/interrupt context helpers.
+ */
+
+#ifndef ZEPHYR_ARCH_ARM_INCLUDE_AARCH32_CORTEX_A_R_EXC_H_
+#define ZEPHYR_ARCH_ARM_INCLUDE_AARCH32_CORTEX_A_R_EXC_H_
+
+#include <arch/cpu.h>
+
+#ifdef _ASMLANGUAGE
+
+/* nothing */
+
+#else
+
+#include <irq_offload.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef CONFIG_IRQ_OFFLOAD
+extern volatile irq_offload_routine_t offload_routine;
+#endif
+
+/* Check the CPSR mode bits to see if we are in IRQ or FIQ mode */
+static ALWAYS_INLINE bool arch_is_in_isr(void)
+{
+	unsigned int status;
+
+	__asm__ volatile(
+			" mrs %0, cpsr"
+			: "=r" (status) : : "memory", "cc");
+	status &= MODE_MASK;
+
+	return	(status == MODE_FIQ) ||
+		(status == MODE_IRQ) ||
+		(status == MODE_SVC);
+}
+
+/**
+ * @brief Setup system exceptions
+ *
+ * Enable fault exceptions.
+ *
+ * @return N/A
+ */
+static ALWAYS_INLINE void z_arm_exc_setup(void)
+{
+}
+
+/**
+ * @brief Clear Fault exceptions
+ *
+ * Clear out exceptions for Mem, Bus, Usage and Hard Faults
+ *
+ * @return N/A
+ */
+static ALWAYS_INLINE void z_arm_clear_faults(void)
+{
+}
+
+extern void z_arm_cortex_r_svc(void);
```

Comment:
why not just z_arm_svc, here? I think similar functions in this header do not have the _cortex_r prefix

Reply:
SVC handling requires some major rework, especially to support user mode.

I will leave this as is for now since it is not related to this PR, and address it when I add userspace support.